vm/src/vm.rs

```diff
@@ -326,19 +331,32 @@ impl GlobalVMState {
         vm
     }
 
-    fn add_types(&self) -> StdResult<(), (TypeId, TcType)> {
+    fn add_types(&mut self) -> StdResult<(), (TypeId, TcType)> {
         use api::generic::A;
         use api::Generic;
+        fn add_type<T: Any>(ids: &mut HashMap<TypeId, TcType>,
+                            env: &mut VMEnv,
+                            name: &str,
+                            typ: TcType) {
+            ids.insert(TypeId::of::<T>(), typ);
+            // Insert aliases so that `find_info` can retrieve information about the primitives
+            env.type_infos.id_to_type.insert(name.into(),
+                                             types::Alias::from(types::AliasData {
+                                                 name: Symbol::new(name),
+                                                 args: Vec::new(),
+                                                 typ: None,
+                                             }));
+        }
+
         {
-            let mut ids = self.typeids.write().unwrap();
-            ids.insert(TypeId::of::<()>(), Type::unit());
-            ids.insert(TypeId::of::<VMInt>(), Type::int());
-            ids.insert(TypeId::of::<i32>(), Type::int());
-            ids.insert(TypeId::of::<u32>(), Type::int());
-            ids.insert(TypeId::of::<u8>(), Type::byte());
-            ids.insert(TypeId::of::<f64>(), Type::float());
-            ids.insert(TypeId::of::<::std::string::String>(), Type::string());
-            ids.insert(TypeId::of::<char>(), Type::char());
+            let ids = self.typeids.get_mut().unwrap();
+            let env = self.env.get_mut().unwrap();
+            add_type::<()>(ids, env, "()", Type::unit());
+            add_type::<VMInt>(ids, env, "Int", Type::int());
+            add_type::<u8>(ids, env, "Byte", Type::byte());
+            add_type::<f64>(ids, env, "Float", Type::float());
+            add_type::<::std::string::String>(ids, env, "String", Type::string());
```

Comment:
The `Value::String` variant is reexported and shadows `std::string::String`. Its a holdover from the time when enum variants existed in the same scope as the enum itself. Haven't gotten around to adding explicit `Value::` prefixes to all places where the enum is used yet.